packages/compile-solidity/test/test_nonrelative.js

```diff
@@ -0,0 +1,61 @@
+const debug = require("debug")("compile:test:test_nonrelative");
+const path = require("path");
+const { Compile } = require("@truffle/compile-solidity");
+const assert = require("assert");
+const Resolver = require("@truffle/resolver");
+const process = require("process");
+
+describe("Non-relative non-absolute file paths", function () {
+  this.timeout(5000); // solc
+
+  const options = {
+    working_directory: __dirname,
+    contracts_directory: path.join(__dirname, "./sources/badSources"),
+    contracts_build_directory: path.join(__dirname, "./does/not/matter"), //nothing is actually written, but resolver demands it
+    compilers: {
+      solc: {
+        version: "0.8.6",
+        settings: {
+          optimizer: {
+            enabled: false,
+            runs: 200
+          }
+        }
+      }
+    },
+    quiet: true
+  };
+  options.resolver = new Resolver(options);
+
+  before("Set working directory", function () {
+    originalWorkingDirectory = process.cwd();
+    process.chdir(options.contracts_directory);
+  });
+
+  it("Refuses to compile non-relative non-absolute paths", async function () {
+    this.timeout(150000);
+    const paths = [
+      "Nonrelative.sol",
+    ].map(filePath => path.join(options.contracts_directory, filePath));
+
+    debug("current dir: %s", process.cwd());
+
+    try {
+      await Compile.sourcesWithDependencies({
+        paths,
+        options
+      });
+      assert.fail("Compilation should have failed");
+    } catch (error) {
+      debug("error: %O", error);
+      if(!error.message.includes('Source "Imported.sol" not found')) {
+        throw error; //rethrow errors that aren't the one we expect
+      }
+      //otherwise, we're good
```

Comment:
Not a nit but a question. Did you use this method for better error output instead of what's available with [assert.rejects](https://nodejs.org/api/assert.html#assert_assert_rejects_asyncfn_error_message)?

Reply:
I suppose this is better anyways because you can assert _why_ it failed to make sure it rejects the way you expect it instead of for some other unexpected reason.

Reply:
That's true @eggplantzzz , that `why` text would certainly be more helpful that `it failed` among a sea of CI output.

Reply:
Honestly, I didn't know about `assert.rejects`.  But I prefer this way anyway for the reason Tyler said.